frontend/src/views/ProjectView.tsx

```diff
@@ -1,3 +1,50 @@
-export default function ProjectView() {
-    return <>Project Management</>;
+import React, { useState } from 'react';
+
+
+import '../styles/CreateProjectDialog.css';
+import { useParams } from 'react-router';
+
+// Define an enum for error types
+enum ErrorType {
+    None = '',
+    PageOpen = 'Failed to open project-view. Please try again.',
+    HardwareError ='No hardware set found.'
 }
+
+
+const ProjectView = () => {
+    const {projectId} = useParams<{projectId: string}>();
+    const [projectName, setProjectName] = useState('');
+    const [description, setDescription] = useState('');
+    const [error, setError] = useState<ErrorType>(ErrorType.None);
+
+
+    const getProjectDetails = () =>{
+        
+    }
+
+    // const handleJoinProject = () => {
+    //     console.log('Joining project');
+    //     // You can add error handling and state updates here
+    // };
```

Comment:
I think that they should be already joined to the project to view it. Otherwise, it's "insecure" or whatever Samant said.